notebooks/utils.py

```diff
@@ -0,0 +1,127 @@
+import torch
+from torch import nn
+from torch import optim
+from matplotlib import pyplot as plt
+import numpy as np
+import seaborn as sns
+
+class Swarm_Simulator:
+    def __init__(self):
+        
+        self.simple_funcs = {
+            'exponential':torch.exp,
+            'sinusoidal':torch.sin
+        }
+        
+        self.debug = False
+        
+        self.params = {
```

Comment:
All 3 of these don't look like they belong to a class instance. Since they look like config, they should be global constants instead.

Reply:
Even params? hmm... interesting. True the simple funcs.

Reply:
Sorry I just pressed a 'start review button', didn't mean to.